### ISSUE TITLE
Allow retrieving more user's best scores 

### ISSUE THREAD

**Issue by stanriders** [ppy/osu-web#7674]
Currently `/users/.../scores/best` allows only the first 100 scores to be retrieved. Since it already supports pagination, it would be good to allow access to more scores, while limiting single query to 100 (if #7673 goes through) scores.

**Comment by peppy** [comment]
Please state the use case for this. Also would this also be something worth displaying on profiles? If not, why?

Basically if an issue hasn't appeared for this until now, the demand is low, so I'd want good reason to warrant this.

**Comment by stanriders** [comment]
Use case would definitely be better profile calculations for `PerformanceCalculator`, since more scores would yield a better final picture and there are usually a lot of scores outside top 100 that are worth looking at. Showing more in profile would also be good for the same reason (a lot of cool scores just aren't high enough pp-wise), but not necessary.

**Comment by peppy** [comment]
We already provide data dumps at https://data.ppy.sh for performance / difficulty calculation purposes. Using that is recommended. Is there a reason that does not suffice?

**Comment by tybug** [comment]
For the record, there are several issues on https://github.com/ppy/osu-api/ for 100+ user best scores:

https://github.com/ppy/osu-api/issues/144, https://github.com/ppy/osu-api/issues/285

**Comment by stanriders** [comment]
Setting up data dumps is a tedious and difficult process, which isn't even supported in `osu-tools`. While it can be done, it's a lot of work, especially for people that only want to use the calculator and not develop something.

**Comment by peppy** [comment]
I'm not sure I understand. Why would someone use the calculator when not developing something? Isn't their PP already calculated on their profile? If possible please try and provide a detailed answer so I don't have to sit here prying teeth out.

Please keep in mind if you don't state a good purpose there is very low chance of the effort being put in to make the infrastructure support this. As mentioned in one of the linked issue, it would require data restructuring to make things efficient enough. Which may come as a bonus from other reorganisation I'm doing for lazer / future but also may not.

**Comment by stanriders** [comment]
People usually use calculator for checking their profiles whenever a new PP rebalance/rework/whatever comes along. Most of the people that are using it usually struggle even with compiling&running the calculator itself so setting up a database is not an option, and demand for recalculated profiles is usually quite big (believe me people dm me a lot asking me to set up a website for whatever new pp rework they currently aware of)

**Comment by peppy** [comment]
I'm not sure that's a valid use case in my eyes. Those people should probably be waiting until we actually look for feedback and make new numbers available.

Will leave this in proposal state. May revisit after we have a new scoring backend live.

**Comment by stanriders** [comment]
Unfortunately you can't just tell people not to look ¯\_(ツ)_/¯ Any new promising pp work always generates A LOT of attention

**Comment by nanaya** [comment]
An unfiltered data* is probably not as taxing but there's currently no useful index for sorting/paginating it.

* as in not excluding scores for same diff with different mods which from what I remember is the most taxing operation on it

**Comment by smoogipoo** [comment]
If you want a good display you need to set up an entire local osu!web, which has its challenges. I still recommend and use my [pp-tester](https://github.com/smoogipoo/osu-web/tree/pp-tester) branch, because https://github.com/ppy/osu-web/issues/5924 has not been fulfilled yet, and without using my branch it's quite hard to navigate through all the [hacks](https://github.com/smoogipoo/osu-web/blob/pp-tester/bin/import_data.sh) required to get a stable deploy.

A good display because without a local osu!web, calculating weighted difficulty is non-trivial if you want well-formatted data. I'm talking about someone that's not a fullstack developer with insight into how the database is set up and who possibly hasn't even used SQL that much because they're mathematicians instead of programmers.  

And when you use a local osu!web you have other challenges to face, for example now you lose the option of debugging unless you re-expose ports via docker (again, non-programmers), figure out that the DB user required to connect to the database is `osuweb` (because I believe `root` dies via the C# mysql connector).

It's quite a daunting task to actually get it all set up for development when you just want to tweak values and try your hand out at implementing some new algorithm.

And no matter how good you have it all set up, you can't compare the relative movement of scores and raw pp values as osu-tools does ([example](https://github.com/ppy/osu-tools/blob/master/PerformanceCalculator/README.md#profile)).

I can go on and on about the disadvantages of using osu-difficulty-calculator. The other option I see is to make osu-tools support the data dumps, but you may not be able to calculate any arbitrary user you want - i.e. let's say I want to recompute my own PP, because I know the relative difficulty of all of my own scores very well, I'm pretty sure I'm not in any of the data dumps.

**Comment by stanriders** [comment]
Going to bump this. 

I'm not sure if anything changed retrieval-expense-wise in 4 years, but it'd still be really nice to allow querying more than 100 scores per user. I think allowing something like 500 would be a perfect option as any more is not really useful (and you better off using full score dump in that case) and it (probably) shouldn't be impossibly expensive to provide

**Comment by peppy** [comment]
If this was going to change, it would first need to change on the display portions (ie user profile) IMO. And since we haven't had a single request for this from a user, I dunno about providing via API alone.

Rationale: the API should not provide more data than is exposed by the game / site.

**Comment by stanriders** [comment]
I agree, I'd expect both to be extended. I think for users that'd be just a "oh nice" thing - it's probably one of those things that if you actually ask the player base everyone would say yes please, but nobody actually cares enough to ask for it

**Comment by peppy** [comment]
@ppy/team-web thoughts on increasing to 200 or 500?

**Comment by MaxOhn** [comment]
Just chiming in to say I'd appreciate being able to fetch more top scores too 👍

My bot's "whatif" command approximates how a user's total pp changes if they got a new given pp value which turned out surprisingly inaccurate when using only 100 top scores so I've been padding an approximation of the top150 instead but using real data would be even better of course.

Same goes for the inverse use-case when a user asks "what pp value am I missing to reach the given total pp" I need to do the same kind of top150 approximation to get better results.

**Comment by nanaya** [comment]
yeah it's probably fine
